pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,390 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from time import sleep
+from time import time as now
+
+from io import BytesIO
+
+import numpy as np
+from pyvisa import VisaIOError
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, truncated_range
+
+
+class HP8753E(Instrument):
+    """Represents the HP8753E Vector Network Analyzer
+    and provides a high-level interface for taking scans of the
+    scattering parameters.
+
+    Keyword arguements:
+    adapter -- <placeholder for description of pymeasure.adapter>
+    name -- <str> describing the instrument
+    """
+
+    def __init__(
+        self,
+        adapter=None,
+        name=None,
+        **kwargs,
+    ):
+        super().__init__(adapter=adapter, name=name, includeSCPI=False, **kwargs)
+
+        self._manu = ""
+        self._model = ""
+        self._fw = ""
+        self._sn = ""
+        self._options = ""
+
+        if name is None:
+            self._manu, self._model, _, self._fw = self.id
+            self._desc = "Vector Network Analyzer"
+            self.name = f"{self._manu} {self._model} {self._desc}"
+        else:
+            self.name = name
+
+    ALLOWED_BANDWIDTH = [10, 30, 100, 300, 1000, 3000, 3700, 6000]
+    SCAN_POINT_VALUES = [3, 11, 21, 26, 51, 101, 201, 401, 801, 1601]
+    SCATTERING_PARAMETERS = ["S11", "S12", "S21", "S22"]
+
+    start_frequency = Instrument.control(
+        "STAR?",
+        "STAR %e Hz",
+        f"""Control the start frequency in Hz. (float truncated from 30_000.0
+        to 6_000_000_000.0).
+        """,
+        validator=truncated_range,
```

Comment:
As said in the other comment, `strict` is better than truncated, as the user will know, when the result differs from the input. With truncation, a user might give `start_frequency=10_000` and believe, that it worked.

Reply:
Switched validator to use `strict_range` instead.